core/field_variable.js

```diff
@@ -81,7 +81,18 @@ Blockly.FieldVariable.prototype.initModel = function() {
   // For instance, some blocks in the toolbox have variable dropdowns filled
   // in by default.
   if (!this.sourceBlock_.isInFlyout) {
-    this.sourceBlock_.workspace.createVariable(this.getValue());
+    // Check if there was exactly one element specified in the
+    // variableTypes list. This is the list that specifies which types of
+    // variables to include in the dropdown.
+    // If there is exactly one element specified, make the variable
+    // being created this specified type. Else, default behavior is to create
+    // a scalar variable
+    if (this.getVariableTypes_().length === 1) {
+      this.sourceBlock_.workspace.createVariable(this.getValue(),
+        this.getVariableTypes_()[0]);
```

Comment:
I expected that you would add an explicit type to the scalar variables ('scalar', maybe).  What's the advantage of leaving it as the empty string?

For context, here's `workspace.createVariable`: 
```
/**
 * Create a variable with a given name, optional type, and optional id.
 * @param {!string} name The name of the variable. This must be unique across
 *     variables and procedures.
 * @param {?string} opt_type The type of the variable like 'int' or 'string'.
 *     Does not need to be unique. Field_variable can filter variables based on
 *     their type. This will default to '' which is a specific type.
 * @param {?string} opt_id The unique id of the variable. This will default to
 *     a UUID.
 * @return {?Blockly.VariableModel} The newly created variable.
 */
Blockly.Workspace.prototype.createVariable = function(name, opt_type, opt_id) {
  if (name.toLowerCase() == Blockly.Variables.noVariableText()) {
    return;
  }
  return this.variableMap_.createVariable(name, opt_type, opt_id);
};
```

Reply:
I agree, we should call out scalar variables specifically instead of leaving the original empty string to specify that type. The only reason that the change hasn't been made in this PR is that it seemed like it would be a very pervasive change and out of the scope of this PR, which is specifically for adding functionality behind the dropdown options for the broadcast messages. I will file an issue for this change.

Reply:
This is now referenced in issue #1251